### ISSUE TITLE
Option for OGCS to wait for Outlook instead of starting it

### ISSUE THREAD

**Issue by yferland** [phw198/OutlookGoogleCalendarSync#535]
**OGCS Version**: v2.7.4.0
**Installed or Portable**: Portable

**Problem Description**  
It would be nice to add an option to choose if we want OGCS to start Outlook if no instance is running (current behavior), or just sit and wait for Outlook to be available.

I have two reasons for this request:
1. I use macros with Outlook, and the way that OGCS starts Outlook (not in full GUI mode), it seems like macros are not available.
1. When I undock my laptop to move from my desk to a meeting room, I need to close Outlook, and then reopen it after I am connected to wifi, otherwise Outlook freezes after a few minutes (probably has to do with using .pst files located on network shares; I know not a good idea), and I don't like the fact that OGCS sometimes restarts Outlook before I am on wifi.

This is why I'd prefer to manually start Outlook when I decide so.

Thank you!

**Comment by leepyswetr** [comment]
I've had the same problem with the macros, and worked around it by starting OGCS from the Outlook startup macro and then killing it on quit.

```
Private Sub Application_Startup()

retval = Shell("C:\Users\xxxx\AppData\Local\OutlookGoogleCalendarSync\OutlookGoogleCalendarSync.exe", vbMinimizedNoFocus)
End Sub

Private Sub Application_Quit()
Dim oServ As Object
Dim cProc As Variant
Dim oProc As Object

Set oServ = GetObject("winmgmts:")
Set cProc = oServ.ExecQuery("Select * from Win32_Process")

For Each oProc In cProc
    If oProc.Name = "OutlookGoogleCalendarSync.exe" Then
      oProc.Terminate (0)
    End If
Next
End Sub
```

**Comment by yferland** [comment]
Thanks for the suggestion. I used your macro and it works great for now!

**Comment by phw198** [comment]
Having started coding something for this*, I'm wondering why this is needed - if OGCS starts Outlook with no GUI (when Outlook's not already running), then it should automatically detach again and the Outlook process should disappear.

The only time it wouldn't do this, is if there's a sync scheduled within the next 5 minutes, otherwise there will be a `INFO` log entry with "Disconnecting from Outlook application." after starting OGCS, or after the initial sync.

So are you saying that you can see an `OUTLOOK.EXE` process that was started by OGCS all the time?

\* I'm not keen on coding a sit-and-wait option, as OGCS needs to connect to Outlook to initialise the application. If it has to wait the only sign OGCS is trying to start is the notification tray icon...

**Comment by phw198** [comment]
Confirmed, this is actually a regression bug...

**Comment by phw198** [comment]
:package: This should now be resolved with [hotfix v2.7.4.9.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/2195722/v2.7.4.9.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)).

When OGCS starts or syncs without Outlook open you should notice the below icon popup in your task tray - and more importantly, now disappear afterwards:
![image](https://user-images.githubusercontent.com/11843015/42734279-3edc1970-8839-11e8-85bf-fc49b0b31289.png)

Please let me know how it goes.

**Comment by phw198** [comment]
I'm going to include this hotfix in the next release - we can reopen the ticket if you still think there's a problem in v2.7.5.
